prow/config.yaml

```diff
@@ -50,15 +50,31 @@ github_reporter:
   - presubmit
   - postsubmit
 
-blocked_branches: &blocked_branches
-  release-*: &blocking_merge
+blocked_branches: &blocked_branches_base
+  master:
+    protect: true
+  release-0.1: &blocking_merge
     required_status_checks:
       contexts:
       - "merges-blocked-needs-admin"
     protect: true
+  release-0.2: *blocking_merge
+  release-0.3: *blocking_merge
+  release-0.4: *blocking_merge
+  release-0.5: *blocking_merge
+  release-0.6: *blocking_merge
+  release-0.7: *blocking_merge
+  release-0.8: *blocking_merge
+  release-1.0: *blocking_merge
+
+blocked_branches: &blocked_branches
+  <<: *blocked_branches_base
+  release-1.1: *blocking_merge
+  release-1.2: *blocking_merge
 
 branch-protection:
```

Comment:
We need to turn on "requires code owners review" for any istio repo other than istio/proxy and istio/istio

Reply:
I presume the behavior when something like required_status_check is not stated for a branch is that whatever is in GitHub is left alone, correct?

Reply:
See the `required_pull_request_reviews` clause defined at the `istio` org level.

All repos will inherit this unless they override it. So right now all istio repos will require the `cla/google` context 
and require 1 an approving review, unless they define otherwise.


Note that [right now](https://github.com/istio/istio/settings/branch_protection_rules/1843352) istio/istio and istio/proxy both requiring these checks on any branches with protection enabled, and this PR is not changing that (are you saying it should?)

Reply:
What I'm asking for is different  than required_pull_request_reviews, we need require_code_owner_reviews: true since we use GitHub's CODEOWNERS files to control who can approve.

Reply:
Ack, wow those are confusingly similar settings.

Can we merge this as is -- it fixes the `release-*` branches -- and I put the `required_code_owner_reviews` changes in the next PR?

These controls exist but we basically don't use them in kubernetes so I don't have experience with them and am therefore less confident in their behavior:
* https://github.com/kubernetes/test-infra/blob/7aef384140833379d3e72f33bc0bb9d10770aefe/prow/config/branch_protection.go#L62-L71
* https://github.com/kubernetes/test-infra/blob/7aef384140833379d3e72f33bc0bb9d10770aefe/prow/config/branch_protection.go#L38-L39

Let me figure out how to express that to the tool and I'll send a PR